src/Sensors/Baro/BMP390.cpp

```diff
@@ -3,46 +3,37 @@
 namespace mmfs
 {
 
-    BMP390::BMP390()
+    BMP390::BMP390(const char *name) : bmp()
     {
-        setName("BMP390");
+        setName(name);
     }
 
-    bool BMP390::initialize()
+    bool BMP390::init()
     {
         if (!bmp.begin_I2C())
         { // hardware I2C mode, can pass in address & alt Wire
             // Serial.println("Could not find a valid BMP390 sensor, check wiring!");
             return initialized = false;
         }
 
-        delay(1000);
+        // delay(1000);
 
         // Set up oversampling and filter initialization
-        bmp.setTemperatureOversampling(BMP3_OVERSAMPLING_8X);
-        bmp.setPressureOversampling(BMP3_OVERSAMPLING_4X);
-        bmp.setIIRFilterCoeff(BMP3_IIR_FILTER_COEFF_3);
-        bmp.setOutputDataRate(BMP3_ODR_50_HZ);
+        int good = 0;
+        good += bmp.setTemperatureOversampling(BMP3_OVERSAMPLING_8X);
+        good += bmp.setPressureOversampling(BMP3_OVERSAMPLING_32X);
+        good += bmp.setIIRFilterCoeff(BMP3_IIR_FILTER_COEFF_3);
+        good += bmp.setOutputDataRate(BMP3_ODR_50_HZ);
 
-        double startPressure = 0;
-        for (int i = 0; i < 10; i++)
-        {
-            bmp.readPressure();
-            delay(25);
-        }
-        for (int i = 0; i < 100; i++)
```

Comment:
Why was the first for loop removed for the 390 but not the 280? Did the initialization procedures end up being different?

Reply:
oversight